src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingService.java

```diff
@@ -415,6 +492,13 @@ public void onConfigChanged() {
             });
 
             mEventProcessor = new EventProcessor(SyncthingService.this, mApi);
+            mEventProcessor.addOnEventBasedModelChangedListener(new EventBasedModel.OnEventBasedModelChangedListener() {
+                @Override
+                public void onEventBasedModelChanged(EventBasedModel model) {
+                    // TODO: needs to run on main thread!
```

Comment:
I think this is still something that I need to clean up, because it used to run on a separate thread and I'm quite sure it is already called in a `Handler` in the meantime.